src/core/WPE.js

```diff
@@ -66,10 +69,16 @@ export default class WPE {
     _noConnectionAfterTime() {
         console.log('_noConnectionAfterTime')
         if (this._state === this.STATES.NOIP) {
-            if (this._wifiPlugin === undefined)
-                this._updateUIState('NoConnection');
-            else
-                this._checkAvailableWifiConfigs()
+            if (this._wifiPlugin === undefined) {
```

Comment:
this means there is no wifi plugin, why are we using a retry if we know there is no wifi on the box?

Reply:
Ok, so is there any mechanism available to differentiate this scenario ? It seems some times the Wifi activation is taking bit longer than usual time, in that case, its leading to 'no valid network connection' error.